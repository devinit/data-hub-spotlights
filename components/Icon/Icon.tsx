import classNames from 'classnames';
import React, { FunctionComponent } from 'react';

interface IconProps {
  name: string;
  className?: string;
}

const Icon: FunctionComponent<IconProps> = ({ className, name }) => (
  <i role="presentation" aria-hidden="true" className={classNames(`ico ico-${name}`, className)}></i>
);

export { Icon };
