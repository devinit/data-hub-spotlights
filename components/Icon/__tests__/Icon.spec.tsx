/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { Icon } from '../Icon';

describe('Icon', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<Icon name="warning-warning" />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders with the specified className', () => {
    const renderer = TestRenderer.create(<Icon name="info-slate" className="ico--12" />).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
