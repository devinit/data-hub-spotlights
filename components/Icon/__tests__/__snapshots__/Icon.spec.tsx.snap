// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<i
  aria-hidden="true"
  className="ico ico-warning-warning"
  role="presentation"
/>
`;

exports[`Icon renders with the specified className 1`] = `
<i
  aria-hidden="true"
  className="ico ico-info-slate ico--12"
  role="presentation"
/>
`;
